testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/maven/metadata/MavenMergeMetadataIT.java

```diff
@@ -0,0 +1,214 @@
+
+import java.io.StringWriter;
+import java.util.Arrays;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.config.Configuration;
+import org.sonatype.nexus.repository.maven.policy.VersionPolicy;
+import org.sonatype.nexus.testsuite.maven.Maven2Client;
+import org.sonatype.nexus.testsuite.maven.MavenITSupport;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Sets;
+import org.apache.http.HttpResponse;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.util.EntityUtils;
+import org.apache.maven.artifact.repository.metadata.Metadata;
+import org.apache.maven.artifact.repository.metadata.Versioning;
+import org.apache.maven.artifact.repository.metadata.io.xpp3.MetadataXpp3Reader;
+import org.apache.maven.artifact.repository.metadata.io.xpp3.MetadataXpp3Writer;
+import org.junit.Before;
+import org.junit.Test;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * Metadata change detection IT.
+ */
+@ExamReactorStrategy(PerClass.class)
+public class MavenMergeMetadataIT
+    extends MavenITSupport
+{
+  private static final String GROUP_ID = "group";
+
+  private static final String ARTIFACT_ID = "artifact";
+
+  private static final String METADATA_PATH = GROUP_ID.replace('.', '/') + "/" + ARTIFACT_ID + "/maven-metadata.xml";
+
+  private static final MetadataXpp3Writer writer = new MetadataXpp3Writer();
+
+  private static final MetadataXpp3Reader reader = new MetadataXpp3Reader();
+
+  private Repository mr1;
+
+  private Repository mr2;
+
+  private Repository mr3;
+
+  private Repository mr4;
+
+  private Repository mavenGroup;
+
+  @Before
+  public void prepare() throws Exception {
+    if ((mr1 = repositoryManager.get("mr1")) == null) {
+      mr1 = repositoryManager.create(hostedConfig("mr1", VersionPolicy.RELEASE));
+    }
+    if ((mr2 = repositoryManager.get("mr2")) == null) {
+      mr2 = repositoryManager.create(hostedConfig("mr2", VersionPolicy.RELEASE));
+    }
+    if ((mr3 = repositoryManager.get("mr3")) == null) {
+      mr3 = repositoryManager.create(hostedConfig("mr3", VersionPolicy.RELEASE));
+    }
+    if ((mr4 = repositoryManager.get("mr4")) == null) {
+      mr4 = repositoryManager.create(hostedConfig("mr4", VersionPolicy.RELEASE));
+    }
+    if (repositoryManager.get("mavenGroup") != null) {
+      repositoryManager.delete("mavenGroup");
+    }
+  }
+
+  private void deployMetadataWithVersionTo(String version, Repository repository) throws Exception {
+    Metadata metadata = get(repository);
```

Comment:
minor, can move the default creation to the 'get' method instead of returning null from there